contrib/win32/win32compat/misc.c

```diff
@@ -746,36 +743,30 @@ w32_rename(const char *old_name, const char *new_name)
 		return -1;
 	
 	/*
-	 * To be consistent with POSIX rename(),
-	 * 1) if the new_name is file, then delete it so that _wrename will succeed.
-	 * 2) if the new_name is directory and it is empty then delete it so that _wrename will succeed.
+	 * To be consistent with POSIX rename(), if the new_name is directory
+	 * and it is empty then delete it so that MoveFileEx will succeed.
 	 */
 	struct _stat64 st_new;
 	struct _stat64 st_old;
 	if ((fileio_stat(new_name, &st_new) != -1) &&
-	    (fileio_stat(old_name, &st_old) != -1)) {
-		if (((st_old.st_mode & _S_IFMT) == _S_IFREG) &&
-		    ((st_new.st_mode & _S_IFMT) == _S_IFREG))
-			w32_unlink(new_name);
-
-		if (((st_old.st_mode & _S_IFMT) == _S_IFDIR) &&
-		    ((st_new.st_mode & _S_IFMT) == _S_IFDIR)) {
+	    (fileio_stat(old_name, &st_old) != -1) &&
+		((st_old.st_mode & _S_IFMT) == _S_IFDIR) &&
+		((st_new.st_mode & _S_IFMT) == _S_IFDIR)) {
 			DIR *dirp = opendir(new_name);
 			if (NULL != dirp) {
 				struct dirent *dp = readdir(dirp);
 				closedir(dirp);
 
 				if (dp == NULL)
 					w32_rmdir(new_name);
-			}
 		}
 	}
 
-	int returnStatus = _wrename(resolvedOldPathName_utf16, resolvedNewPathName_utf16);
+	const int returnStatus = MoveFileExW(resolvedOldPathName_utf16, resolvedNewPathName_utf16, MOVEFILE_REPLACE_EXISTING | MOVEFILE_COPY_ALLOWED);
 	free(resolvedOldPathName_utf16);
 	free(resolvedNewPathName_utf16);
-
-	return returnStatus;
+	
+	return (returnStatus == 0) ? (errno_from_Win32LastError()) : 0;
```

Comment:
The [caller](https://github.com/PowerShell/openssh-portable/blob/901f123915e5b6b9cb25a3d65537318a766c73c7/sftp-server.c#L1318) expects this function to return -1 incase of error and reads the error from errno variable.

Reply:
Good catch.  I believe I fixed it up.